sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/HttpUrlConnectionAsyncClient.java

```diff
@@ -0,0 +1,206 @@
+                                        progressReporter.reportProgress(buffer.remaining());
+                                        return buffer;
+                                    });
+                                }
+
+                                requestBody
+                                    .flatMap(buffer -> {
+                                        try {
+                                            byte[] bytes = new byte[buffer.remaining()];
+                                            buffer.get(bytes);
+                                            os.write(bytes);
+                                            return Mono.just(buffer); // Emit the buffer for downstream processing if needed
+                                        } catch (IOException e) {
+                                            return FluxUtil.monoError(LOGGER, new RuntimeException(e));
+                                        }
+                                    })
+                                    .then()
+                                    .block(); // Wait for completion of the write operations
+                                os.flush();
+                            } catch (IOException e) {
+                                return FluxUtil.monoError(LOGGER, new RuntimeException(e));
+                            }
+                        }
+                        break;
+                    case GET:
+                    case HEAD:
+                    case OPTIONS:
+                    case TRACE:
+                    case CONNECT:
+                        break;
+                    default:
+                        return FluxUtil.monoError(LOGGER, new IllegalStateException("Unknown HTTP Method:"
+                            + httpRequest.getHttpMethod()));
+                }
+
+                // Read response
+                int responseCode = connection.getResponseCode();
+
+                Map<String, List<String>> responseHeadersMap = new HashMap<>();
+                for (Map.Entry<String, List<String>> entry : connection.getHeaderFields().entrySet()) {
+                    if (entry.getKey() != null) {
+                        responseHeadersMap.put(entry.getKey(), entry.getValue());
+                    }
+                }
+
+                ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                try (InputStream errorStream = connection.getErrorStream()) {
+                    InputStream inputStream = (errorStream == null) ? connection.getInputStream() : errorStream;
+                    byte[] buffer = new byte[1024];
+                    int length;
+                    while ((length = inputStream.read(buffer)) != -1) {
+                        outputStream.write(buffer, 0, length);
+                    }
+                }
+
+                connection.disconnect();
+
+                return Mono.just(new HttpUrlConnectionResponse(
```

Comment:
Since this API is writing to blocking streams and reading from blocking streams, this should be `publishOn(Schedulers.boundedElastic())` to prevent blocking non-blocking threads.